apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx

```diff
@@ -0,0 +1,156 @@
+  const { colors } = useTheme();
+  const [device, setDevice] = useState<Device | null>(null);
+  const [stepIndex, setStepIndex] = useState<number>(0);
+
+  const { onboardingState, allowedError, fatalError } = useOnboardingStatePolling({ device, pollingPeriodMs });
+
+  const { pairedDevice } = route.params; 
+
+  // When reaching the synchronized onboarding, the device could already
+  // be BLE paired to the phone and at the same time not known by LLM,
+  // it does not matter.
+  // So as long as the device is not onboarded, the LLM will ask for a pairing.
+  useEffect(() => {
+    console.log("SyncOnboarding: navigate to pairDevices");
+
+    // @ts-expect-error navigation issue
+    navigation.navigate(ScreenName.PairDevices, {
+      onlySelectDeviceWithoutFullAppPairing: true,
+      onDoneNavigateTo: ScreenName.SyncOnboardingWelcome,
+    });
+  }, [navigation]);
+
+  const handleOnPaired = useCallback((pairedDevice: Device) => {
+    console.log(
+      `SyncOnboarding: handleOnPaired ${JSON.stringify(pairedDevice)}`,
+    );
+
+    setDevice(pairedDevice);
+  }, []);
+
+  // Triggered when a new paired device is passed when navigating to this screen
+  // It avoids having a callback function passed to the PairDevices screen
+  useEffect(() => {
+    if (pairedDevice) {
+      handleOnPaired(pairedDevice);
+    }
+  }, [pairedDevice, handleOnPaired]);
+
+  // WIP: only for demo
+  const onboardingSteps = [
+    "Pairing device",
+    "Welcome Page",
+    "Setup choice",
+    "Setting up pin",
+    `Writing seed words ${
+      onboardingState && onboardingState.currentOnboardingStep === OnboardingStep.NewDevice
+        ? onboardingState.currentSeedWordIndex + 1
+        : ""
+    }`,
+    `Confirming seed words ${
+      onboardingState && onboardingState.currentOnboardingStep === OnboardingStep.NewDeviceConfirming
+        ? onboardingState.currentSeedWordIndex + 1
+        : ""
+    }`,
+    "Safety Warning",
+    "Ready ?"
+  ];
+
+  // Updates UI step index from the onboarding state
+  useEffect(() => {
+    if (!device) {
+      // No device is paired yet
+      setStepIndex(0);
+      return;
+    }
+
+    // No change if the onboardingState is null
+    if (!onboardingState) {
+      return;
+    }
+    
+    switch(onboardingState?.currentOnboardingStep) {
+      case OnboardingStep.WelcomeScreen:
+        setStepIndex(1);
+        break;
+      case OnboardingStep.SetupChoice:
+        setStepIndex(2);
+        break;
+      case OnboardingStep.Pin:
+        setStepIndex(3);
+        break;
+      case OnboardingStep.NewDevice:
+        setStepIndex(4);
+        break;
+      case OnboardingStep.NewDeviceConfirming:
+        setStepIndex(5);
+        break;
+      case OnboardingStep.SafetyWarning:
+        setStepIndex(6);
+        break;
+      case OnboardingStep.Ready:
+        setStepIndex(7);
+        break;
+      default:
+        setStepIndex(0);
```

Comment:
there may be a way to simplify this with an array containing all the steps that you are currently switch()ing on.

```js
const steps = [OnboardingStep.WelcomeScreen, ...]
const stepIndex = steps.indexOf(onboardingState?.currentOnboardingStep) + 1
```

Reply:
Yeah, definitely I'll try to make something simpler like this ! Right now for this screen it was to demo and show that the polling mechanism was working ;)

Reply:
+1